File: bin/model/Forecast.py
from model.Interval import Interval
from model.Source import Source
import datetime


class Forecast:

    __expected_attributes = [
        ("source", Source),
        ("ticker", str),
        ("interval", Interval),
        ("timestamp", datetime),
        ("price", float),
    ]

    def __init__(self, values: dict) -> None:
        for key in self.__expected_attributes:
            if key[0] in values and isinstance(values[key[0]], key[1]):
                setattr(self, key[0], values[key[0]])
            else:
                setattr(self, key[0], None)

File: api/src/main/java/com/masterson/api/model/forecast/Forecast.java
package com.masterson.api.model.forecast;

import com.masterson.api.model.BaseEntity;
import com.masterson.api.model.Interval;
import com.masterson.api.model.Source;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.Id;
import javax.persistence.IdClass;

@Entity(name = "FORECAST")
@IdClass(ForecastId.class)
@Getter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Forecast extends BaseEntity {

    @Id
    @Column(name = "SOURCE")
    private Source source;

    @Id
    @Column(name = "TICKER")
    private String ticker;

    @Id
    @Column(name = "TIME_INTERVAL")
    private Interval interval;

    @Id
    @Column(name = "FUTURE_TIME")
    private String futureTime;

    @Column(name = "PRICE")
    private float price;

    @Column(name = "PRICE_C")
    private float correctPrice;
}
